```bash
WARNINGS:
?: (axes.W001) You are using the django-axes cache handler for login attempt tracking. Your cache configuration is however invalid and will not work correctly with django-axes. This can leave security holes in your login systems as attempts are not tracked correctly. Reconfigure settings.AXES_CACHE and settings.CACHES per django-axes configuration documentation.
Traceback (most recent call last):
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 362, in _get_applied_migrations_and_create_tables
    return self._get_applied_migrations(migrations_package_name, replicated)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 379, in _get_applied_migrations
    return set(obj.module_name for obj in self.select(query))
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 379, in <genexpr>
    return set(obj.module_name for obj in self.select(query))
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 276, in select
    r = self._send(query, settings, True)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 389, in _send
    raise ServerError(r.text)
infi.clickhouse_orm.database.ServerError: Code: 60. DB::Exception: Table posthog_test.infi_clickhouse_orm_migrations_distributed doesn't exist. (UNKNOWN_TABLE) (version 23.6.1.1524 (official build))
 (0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/luna/Documents/posthog/manage.py", line 21, in <module>
    main()
  File "/Users/luna/Documents/posthog/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/Users/luna/Documents/posthog/posthog/management/commands/migrate_clickhouse.py", line 36, in handle
    self.migrate(CLICKHOUSE_HTTP_URL, options)
  File "/Users/luna/Documents/posthog/posthog/management/commands/migrate_clickhouse.py", line 74, in migrate
    database.migrate(MIGRATIONS_PACKAGE_NAME, options["upto"], replicated=True)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 346, in migrate
    applied_migrations = self._get_applied_migrations_and_create_tables(migrations_package_name, replicated=replicated)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 366, in _get_applied_migrations_and_create_tables
    self.create_table(MigrationHistoryReplicated)
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 152, in create_table
    self._send(model_class.create_table_sql(self))
  File "/Users/luna/Documents/posthog/venv/lib/python3.10/site-packages/infi/clickhouse_orm/database.py", line 389, in _send
    raise ServerError(r.text)
infi.clickhouse_orm.database.ServerError: Code: 253. DB::Exception: There was an error on [localhost:9000]: Code: 253. DB::Exception: Replica /clickhouse/prod/tables/noshard/posthog_test/infi_clickhouse_orm_migrations/replicas/ch1-01 already exists. (REPLICA_ALREADY_EXISTS) (version 23.6.1.1524 (official build)). (REPLICA_ALREADY_EXISTS) (version 23.6.1.1524 (official build))
 (0)
[1]    32151 terminated  ./bin/e2e-test-runner

```



I thought I got that right yesterday. But when I try to execute the same script (`e2e-test-runner`), I ran into this error. Even if this didn't show up and I try to run tests in cypress app, I get 400 for `/api/login`. How to correctly run that `e2e-test-runner` script and not run into these DB related issues?

--- a/cypress/e2e/early-access-management.cy.ts
+++ b/cypress/e2e/early-access-management.cy.ts
@@ -1,0 +1,34 @@
+describe('Early Access Management', () => {
+    beforeEach(() => {
+        cy.visit('/early_access_features')
+    })
+
+    it('Early access feature new and list', () => {
+        // load an empty early access feature page
+        cy.get('h1').should('contain', 'Early Access Management')
+        cy.title().should('equal', 'Early Access Management • PostHog')
+        cy.get('h2').should('contain', 'Create your first feature')
+        cy.get('[data-attr="product-introduction-docs-link"]').should(
+            'contain',
+            'Learn more about Early access features'
+        )
+
+        // go to create a new feature
+        cy.get('[data-attr="create-feature"]').click()
+
+        // New Feature Release page
+        cy.get('h1').should('contain', 'New Feature Release')
+
+        // set feature name & description
+        cy.get('[data-attr="feature-name"]').type('Test Feature')
+
+        // save
+        cy.get('[data-attr="save-feature"]').click()
+        cy.get('[data-attr=success-toast]').contains('Early Access Feature saved').should('exist')
+
+        // back to features
+        cy.visit('/early_access_features')
+        cy.get('tbody').contains('Test Feature')
+        cy.get('h2').should('not.have.text', 'Create your first feature')
+    })
+})

--- a/cypress/e2e/early-access-management.cy.ts
+++ b/cypress/e2e/early-access-management.cy.ts
@@ -19,8 +19,14 @@ describe('Early Access Management', () => {
         // New Feature Release page
         cy.get('h1').should('contain', 'New Feature Release')
 
+        // cancel new feature
+        cy.get('[data-attr="cancel-feature"]').click()
+        cy.get('h1').should('contain', 'Early Access Management')
+
         // set feature name & description
+        cy.get('[data-attr="create-feature"]').click()
         cy.get('[data-attr="feature-name"]').type('Test Feature')
+        cy.get('[data-attr="save-feature').should('contain.text', 'Save as draft')
 
         // save
         cy.get('[data-attr="save-feature"]').click()
@@ -30,5 +36,20 @@ describe('Early Access Management', () => {
         cy.visit('/early_access_features')
         cy.get('tbody').contains('Test Feature')
         cy.get('h2').should('not.have.text', 'Create your first feature')
+
+        // edit feature
+        cy.get('a.Link').contains('.row-name', 'Test Feature').click()
+        cy.get('[data-attr="edit-feature"]').click()
+        cy.get('h1').should('contain', 'Test Feature')
+        cy.get('[data-attr="save-feature"]').should('contain.text', 'Save')
+
+        // delete feature
+        cy.get('[data-attr="save-feature"]').click()
+        cy.get('[data-attr="delete-feature"]').click()
+        cy.get('h3').should('contain', 'Permanently delete feature?')
+        cy.get('[data-attr="confirm-delete-feature"]').click()
+        cy.get('[data-attr=info-toast]')
+            .contains('Early access feature deleted. Remember to delete corresponding feature flag if necessary')
+            .should('exist')
     })
 })

--- a/frontend/src/scenes/early-access-features/EarlyAccessFeature.tsx
+++ b/frontend/src/scenes/early-access-features/EarlyAccessFeature.tsx
@@ -117,6 +117,7 @@ export function EarlyAccessFeature({ id }: { id?: string } = {}): JSX.Element {
                                                 children: 'Delete',
                                                 type: 'primary',
                                                 status: 'danger',
+                                                'data-attr': 'confirm-delete-feature',
                                                 onClick: () => {
                                                     // conditional above ensures earlyAccessFeature is not NewEarlyAccessFeature
                                                     deleteEarlyAccessFeature(
@@ -162,7 +163,12 @@ export function EarlyAccessFeature({ id }: { id?: string } = {}): JSX.Element {
                                 )}
                                 <LemonDivider vertical />
                                 {earlyAccessFeature.stage != EarlyAccessFeatureStage.GeneralAvailability && (
-                                    <LemonButton type="secondary" onClick={() => editFeature(true)} loading={false}>
+                                    <LemonButton
+                                        type="secondary"
+                                        onClick={() => editFeature(true)}
+                                        loading={false}
+                                        data-attr="edit-feature"
+                                    >
                                         Edit
                                     </LemonButton>
                                 )}

--- a/frontend/src/scenes/early-access-features/EarlyAccessFeature.tsx
+++ b/frontend/src/scenes/early-access-features/EarlyAccessFeature.tsx
@@ -77,6 +77,7 @@ export function EarlyAccessFeature({ id }: { id?: string } = {}): JSX.Element {
                             <>
                                 <LemonButton
                                     type="secondary"
+                                    data-attr="cancel-feature"
                                     onClick={() => {
                                         if (isEditingFeature) {
                                             editFeature(false)
@@ -92,6 +93,7 @@ export function EarlyAccessFeature({ id }: { id?: string } = {}): JSX.Element {
                                 <LemonButton
                                     type="primary"
                                     htmlType="submit"
+                                    data-attr="save-feature"
                                     onClick={() => {
                                         submitEarlyAccessFeatureRequest(earlyAccessFeature)
                                     }}

